### DISCUSSION TITLE
Unix Protections on .gcm Dir

### DISCUSSION THREAD

**Discussion by BenAigan** [git-ecosystem/git-credential-manager#393] [Ideas]
Is it possible for the Git Credential Manager Core to create a directory with user only permissions rather than read and execute for the rest of the world especially if using plain text Credential Store?

**Comment by mjcheetham** [comment]
Hello!

From https://github.com/microsoft/Git-Credential-Manager-Core/blob/3f6b956/docs/linuxcredstores.md#4-plaintext-files you can see that we recommend that the plaintext credential store directory should be set to only be readable by the owner/user and not others:

> If you chose to use this credential store, it is recommended you set the permissions on this directory such that no other users or applications can access files within. If possible, use a path that exists on an external volume that you take with you and use full-disk encryption.

Right now it's on the user to `chmod` the default directory (`$HOME/.gcm/store`) to `700`, but it's [a reasonable ask](396) to have GCM create the directory with the appropriate permissions if it does not exist already (and not modify any existing directory).

**Reply by mjcheetham** [reply]
@benaigan this feature has now been added and released in version [2.0.498](https://github.com/microsoft/Git-Credential-Manager-Core/releases/tag/v2.0.498).
